hudi-client/src/main/java/org/apache/hudi/table/HoodieCopyOnWriteTable.java

```diff
@@ -602,18 +602,39 @@ private int addUpdateBucket(String fileIdHint) {
       return bucket;
     }
 
-    private void assignInserts(WorkloadProfile profile) {
+    private void assignInserts(WorkloadProfile profile, JavaSparkContext jsc) {
       // for new inserts, compute buckets depending on how many records we have for each partition
       Set<String> partitionPaths = profile.getPartitionPaths();
       long averageRecordSize =
           averageBytesPerRecord(metaClient.getActiveTimeline().getCommitTimeline().filterCompletedInstants(),
               config.getCopyOnWriteRecordSizeEstimate());
       LOG.info("AvgRecordSize => " + averageRecordSize);
+
+      HashMap<String, List<SmallFile>> partitionSmallFilesMap = new HashMap<>();
+      if (jsc != null && partitionPaths.size() > 1) {
```

Comment:
I think its a reasonable thing to parallelize this.. Listing of cleaning etc has been parallelized like this before. Should be safe to do. 

Also can we pull this block into a method? `getSmallFiles(partitionPaths)`

Reply:
good point, will move to method like getSmallFilesForPartitions(partitionPaths, jsc)